src/main/scala/io/iohk/ethereum/ledger/BlockImport.scala

```diff
@@ -197,8 +195,14 @@ class BlockImport(
       blockchain.save(block, receipts, td, saveAsBestBlock = false)
     }
 
+    val checkpointNumber = oldBranch
+      .filter(_.block.hasCheckpoint)
+      .sortBy(_.block.number)
+      .lastOption
```

Comment:
Minor: instead of sortBy and lastOption, should we maybe use max or maxBy?

Reply:
that was my first option, but maxBy/max are always returning value - in this case, we could have an empty list.

Reply:
Yeah, that's a pain of usage, but maybe we could have done `Try(oldBranch.collect { case b if b.block.hasCheckpoint => b.block.number }).max).toOption`?

Reply:
IMHO your proposition is less readable, but it's probably doing one less iteration (filter, sortBy, lastOption vs collect, max). WDYT about `.filter(_.block.hasCheckpoint).sortBy(-_.block.number).headOption` ?

Reply:
What about https://typelevel.org/cats/api/cats/Foldable.html#maximumByOption[A,B](fa:F[A])(f:A=%3EB)(implicitevidence$3:cats.Order[B]):Option[A] ?